sweetest/src/main/java/com/mysugr/sweetest/framework/context/DependenciesTestContext.kt

```diff
@@ -83,7 +99,35 @@ class DependenciesTestContext {
      *  [provide]. [provide] automatically requires that the automatically created instance is used.
      */
     fun provide(clazz: KClass<*>) {
-        val dependency = getDependencyConfiguration(clazz)
-        TestEnvironment.dependencies.states[dependency].mode = DependencyMode.REAL
+        val mode = DependencyMode.REAL
+        addDependency(clazz) { state, _ ->
+            state.mode = mode
+        }
+    }
+
+    /**
+     * Checks if an instance of [clazz] is already added via the global module configuration.
+     * If so, the found [DependencyConfiguration] is added to the [DependencyManager.states].
+     * If not, a [DependencyConfiguration] is created dynamically and also added.
+     */
+    private fun addDependency(clazz: KClass<*>, mode: DependencyMode? = null, block: (DependencyState<out Any>, DependencyMode?) -> Unit) {
+        with(TestEnvironment.dependencies) {
+            val dependencyState = getDependencyConfiguration(clazz)?.let { configuration ->
+                states[configuration]
+            } ?: run {
+                states[DependencyConfiguration(clazz, null, null, mode)]
+            }
+
+            block(dependencyState, dependencyState.configuration.defaultDependencyMode)
+        }
+    }
+
+    /**
+     *
+     */
+    private fun assertMode(clazz: KClass<*>, configurationMode: DependencyMode?, mode: DependencyMode) {
```

Comment:
this checks for the mode change compared to the configuration but not whether the mode has changed from the last time somebody has enforced a mode. i'll write a test for it and see what it does...

Reply:
If the mode is set repeatedly the it throws either way. 
```
          if (value != modeField) {
                if (modeField != null) {
                    throw IllegalStateException(
                        "Can't change dependency mode or \"${configuration.clazz}\", it " +
                            "has already been set before and can't be changed afterwards"
                    )
                }
```